src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -49,6 +60,57 @@ internal bool AutomaticRedirection
                 _automaticRedirection = value;
             }
         }
+        internal bool UseProxy
+        {
+            get
+            {
+                if (_proxyPolicy == CurlProxyUsePolicy.UseDefaultProxy)
+                {
+                    return false;
+                }
+                return _proxyPolicy != CurlProxyUsePolicy.DoNotUseProxy;
```

Comment:
I'll change the value of UseProxy to return true when the state is defaultProxy. 

However,  the code as it is now, is indeed doing exactly as your spec suggests. The proxy settings passed to libcurl in SendAsync are based, **only** on the value of ProxyPolicy (state contains Policy field passed explicitly. And only that is used to determine the value).  SendAsync does not look at the value of UsePolicy.  
 
Do you suggest I change the code in the SendAsync too to use UseProxy property so that the code "looks" like the algorithm you described? I believe using a tri-valued variable like I did makes it more clear but I  could be wrong.